doc/osh-help.md

```diff
@@ -805,6 +806,135 @@ these are discouraged.
 
 <!--    -R VAR     True if the variable VAR has been set and is a nameref variable. -->
 
+<a h4="printf">printf</h4>
```

Comment:
These can all use markdown now:

```
#### printf
````

I changed the toolchain to support that.

(Also there is a typo here -- should be h4 tag, not a tag!)

Reply:
Sorry for the delay! Here is a new version
 * I added a new topic for the character escapes. I am afraid I wasn't able to find more differences between the echo and printf escapes, the code was too obscure for me.
 * I tried to fix the getopts "errors mode" part, but I'm not sure I did it right, this command is pretty weird and I wasn't able to debug the osh command
 * All the other suggestions were incorporated.